src/mon/PaxosService.cc

```diff
@@ -382,20 +382,39 @@ void PaxosService::maybe_trim()
     return;
 
   version_t to_remove = trim_to - get_first_committed();
-  if (g_conf()->paxos_service_trim_min > 0 &&
-      to_remove < (version_t)g_conf()->paxos_service_trim_min) {
+  version_t paxos_service_trim_min = (version_t)g_conf()->paxos_service_trim_min;
+  version_t paxos_service_trim_max = (version_t)g_conf()->paxos_service_trim_max;
+  int paxos_service_trim_max_multiplier = g_conf()->paxos_service_trim_max_multiplier;
+
+  if (paxos_service_trim_max_multiplier) {
+    paxos_service_trim_max = paxos_service_trim_max * paxos_service_trim_max_multiplier;
+  }
+
+  if (paxos_service_trim_min > 0 &&
+      to_remove < paxos_service_trim_min) {
     dout(10) << __func__ << " trim_to " << trim_to << " would only trim " << to_remove
-	     << " < paxos_service_trim_min " << g_conf()->paxos_service_trim_min << dendl;
+	     << " < paxos_service_trim_min " << paxos_service_trim_min << dendl;
     return;
   }
 
-  if (g_conf()->paxos_service_trim_max > 0 &&
-      to_remove > (version_t)g_conf()->paxos_service_trim_max) {
-    dout(10) << __func__ << " trim_to " << trim_to << " would only trim " << to_remove
-	     << " > paxos_service_trim_max, limiting to " << g_conf()->paxos_service_trim_max
+  if (paxos_service_trim_max > 0 &&
+      to_remove > paxos_service_trim_max ) {
+    version_t trim_difference = to_remove - paxos_service_trim_max;
+    if (trim_difference > paxos_service_trim_max * 0.5 && !paxos_service_trim_max_multiplier) {
+      
+      //Increasing trim_max limit when to_remove is big
+      
+      dout(10) << __func__ << " changing paxos service trim max " << dendl;
+      paxos_service_trim_max = paxos_service_trim_max + (trim_difference/2);
```

Comment:
the multiplier is working a bit differently than I expected. I was thinking it would affect the upper bound of adjustment we would make to paxos_service_trim_max, e.g.:

```
paxos_service_trim_max = min(paxos_service_trim_max * paxos_serivce_trim_max_multiplier, trim_difference)
```

since paxos_service_trim_max is a local variable here, reset each time we trim, it's not as dynamic - it seems equivalent to increasing the default paxos_service_trim_max. Could you include a comment describing the algorithm? Maybe I'm misunderstanding the intent.

Reply:
nevermind the 2nd part, I think I get the algorithm now. since it's depending on trim_difference it will lead to higher rates of trimming when there are higher input rates.

Reply:
Okay got it. Yes that makes more sense here since we can still dynamically change the max but only to a certain limit. 
I'll make the change.